esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5885,4 +5897,207 @@ int CWsDfuEx::GetIndexData(IEspContext &context, bool bSchemaOnly, const char* i
     return iRet;
 }
 
+unsigned CWsDfuEx::getFilePartsInfo(IEspContext &context, IDistributedFile *df, const char *clusterName,
+    IArrayOf<IEspDFUPartLocations> &dfuPartLocations, IArrayOf<IEspDFUPartCopies> &dfuPartCopies)
+{
+    int nextLocationsIndex = 0;
```

Comment:
@wangkx @rpastrana - I think these should be 1 based for consistency (PartId's are in same response structure).

Reply:
@wangkx  and I discussed this offline. We acknowledged the possible confusion, but ultimately chose this approach because we're dealing with 'ip locations' found via an index to an array, and file part ids (not necessarily a file part index). Unless we're inferring file id based on the XML array position (which we should prob stay away from)

Reply:
Well, IMHO it's inconsistent and confusing to anyone consuming the message.
The part ids aren't id's in any fabricated unique sense they are precisely the part index.